packages/react-components/src/components/VideoGallery/LargeGalleryLayout.tsx

```diff
@@ -0,0 +1,185 @@
+
+  // This is for tracking the number of children in the first page of overflow gallery.
+  // This number will be used for the maxOverflowGalleryDominantSpeakers when organizing the remote participants.
+  // We need to add the local participant to the pinned participant count so we are placing the speakers correctly.
+  const childrenPerPage = useRef(4);
+  const { gridParticipants, overflowGalleryParticipants } = useOrganizedParticipants({
+    remoteParticipants,
+    localParticipant,
+    dominantSpeakers,
+    maxRemoteVideoStreams: maxStreamsTrampoline(),
+    isScreenShareActive: !!screenShareComponent,
+    maxOverflowGalleryDominantSpeakers: screenShareComponent
+      ? childrenPerPage.current - ((pinnedParticipantUserIds.length + 1) % childrenPerPage.current)
+      : childrenPerPage.current,
+    /* @conditional-compile-remove(pinned-participants) */ pinnedParticipantUserIds,
+    /* @conditional-compile-remove(gallery-layouts) */ layout: 'largeGallery'
+  });
+  console.log('gridParticipants', gridParticipants);
+  let activeVideoStreams = 0;
+
+  const gridTiles = gridParticipants.map((p) => {
+    return onRenderRemoteParticipant(
+      p,
+      maxRemoteVideoStreams && maxRemoteVideoStreams >= 0
+        ? p.videoStream?.isAvailable && activeVideoStreams++ < maxRemoteVideoStreams
+        : p.videoStream?.isAvailable
+    );
+  });
+
+  /**
+   * instantiate indexes available to render with indexes available that would be on first page
+   *
+   * For some components which do not strictly follow the order of the array, we might
+   * re-render the initial tiles -> dispose them -> create new tiles, we need to take care of
+   * this case when those components are here
+   */
+  const [indexesToRender, setIndexesToRender] = useState<number[]>([]);
+
+  const overflowGalleryTiles = overflowGalleryParticipants.map((p, i) => {
+    return onRenderRemoteParticipant(
+      p,
+      maxRemoteVideoStreams && maxRemoteVideoStreams >= 0
+        ? p.videoStream?.isAvailable && indexesToRender.includes(i) && activeVideoStreams++ < maxRemoteVideoStreams
+        : p.videoStream?.isAvailable
+    );
+  });
+
+  if (localVideoComponent) {
+    gridTiles.push(localVideoComponent);
+  }
+
+  const overflowGallery = useMemo(() => {
+    if (overflowGalleryTiles.length === 0) {
+      return null;
+    }
+    return (
+      <OverflowGallery
+        isNarrow={isNarrow}
+        /* @conditional-compile-remove(gallery-layouts) */
+        isShort={isShort}
+        shouldFloatLocalVideo={false}
+        overflowGalleryElements={overflowGalleryTiles}
+        horizontalGalleryStyles={styles?.horizontalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        verticalGalleryStyles={styles?.verticalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        overflowGalleryPosition={overflowGalleryPosition}
+        onFetchTilesToRender={setIndexesToRender}
+        onChildrenPerPageChange={(n: number) => {
```

Comment:
nit: pass in memoized fn

Reply:
if others have it and we see no issue, might be OK to leave it, will leave this up to you